src/main/java/org/nrnb/gsoc/enrichment/tasks/FilterEnrichmentTableTask.java

```diff
@@ -16,92 +17,114 @@
     private CyServiceRegistrar registrar;
     private List<TermSource> categoryFilter = TermSource.getValues();
-	private final Logger logger = Logger.getLogger(CyUserLog.NAME);
-
-
-	@Tunable(description = "Select categories",
-			tooltip = "Select the enrichment categories to show in the table",
-			longDescription = "Select the enrichment categories to show in the table. Use \"All\" to remove the filtering.",
-			exampleStringValue = "GO Process",
-			gravity = 1.0)
-	public ListMultipleSelection<TermSource> categories = new ListMultipleSelection<>(TermSource.getValues());
-
-	//public boolean removeOverlapping = false;
-
-	//public BoundedDouble overlapCutoff = new BoundedDouble(0.0, 0.5, 1.0, false, false);
-
-	public FilterEnrichmentTableTask(final CyServiceRegistrar registrar, EnrichmentCytoPanel panel) {
-		this.registrar = registrar;
-		applicationManager = registrar.getService(CyApplicationManager.class);
-		this.network = applicationManager.getCurrentNetwork();
-		this.enrichmentPanel = panel;
-	}
-
-
-
-	@Override
-	public void run(TaskMonitor monitor) throws Exception {
-		monitor.setTitle("Filter Enrichment table");
-		logger.info("Filtering results based on selected value: " + categories.getSelectedValues());
-
-		List<TermSource> categoryList = categories.getSelectedValues();
-		if (enrichmentPanel == null) {
-			CySwingApplication swingApplication = registrar.getService(CySwingApplication.class);
-			CytoPanel cytoPanel = swingApplication.getCytoPanel(CytoPanelName.SOUTH);
-			if (cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment") != -1)
-				enrichmentPanel = (EnrichmentCytoPanel) cytoPanel.getComponentAt(
-						cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment"));
-			else return;
-		}
-		EnrichmentTableModel tableModel = enrichmentPanel.getTableModel();
-		if (tableModel == null){
-			monitor.showMessage(TaskMonitor.Level.ERROR,  "Unable to find enrichment table!");
-		}
-		tableModel.filter(categoryList);
-		logger.info("Filtering results completed");
-	}
-
-	@Override
-	@SuppressWarnings("unchecked")
-	public <R> R getResults(Class<? extends R> clzz) {
-		if (clzz.equals(String.class)) {
-			if (filteredEnrichmentTable != null)
-				return (R)("\"EnrichmentTable\": "+filteredEnrichmentTable.getSUID());
-			return (R)"";
-		} else if (clzz.equals(JSONResult.class)) {
-			JSONResult res = () -> {
-				if (filteredEnrichmentTable != null)
-					return "{\"EnrichmentTable\": "+filteredEnrichmentTable.getSUID()+"}";
-				return "{}";
-			};
-			return (R)res;
-		}
-		return null;
-	}
-
-	@Override
-	public List<Class<?>> getResultClasses() {
-		return Arrays.asList(JSONResult.class, String.class);
-	}
-
-	@ProvidesTitle
-	public String getTitle() {
-		return "Filter Enrichment table";
-	}
+    private final Logger logger = Logger.getLogger(CyUserLog.NAME);
+
+    //public boolean removeOverlapping = false;
+
+    //public BoundedDouble overlapCutoff = new BoundedDouble(0.0, 0.5, 1.0, false, false);
```

Comment:
Remove the code if you are not using it.

Reply:
Done